packages/next-swc/crates/napi/src/lib.rs

```diff
@@ -73,19 +76,84 @@ shadow_rs::shadow!(build);
 static ALLOC: turbopack_binding::turbo::malloc::TurboMalloc =
     turbopack_binding::turbo::malloc::TurboMalloc;
 
+static LOG_THROTTLE: Mutex<Option<Instant>> = Mutex::new(None);
+static LOG_FILE_PATH: &str = ".next/turbopack.log";
+
 #[cfg(feature = "__internal_dhat-heap")]
 #[global_allocator]
 static ALLOC: dhat::Alloc = dhat::Alloc;
 
 #[cfg(not(target_arch = "wasm32"))]
 #[napi::module_init]
+
 fn init() {
-    if cfg!(debug_assertions) || env::var("SWC_DEBUG").unwrap_or_default() == "1" {
-        set_hook(Box::new(|panic_info| {
-            let backtrace = Backtrace::new();
-            println!("Panic: {:?}\nBacktrace: {:?}", panic_info, backtrace);
-        }));
-    }
+    use std::{fs::OpenOptions, io};
+
+    set_hook(Box::new(|panic_info| {
+        let mut last_error_time = LOG_THROTTLE.lock().unwrap();
```

Comment:
Thanks for changing this. Some comment might be useful here?

```suggestion
        // hold open this mutex guard to prevent concurrent writes to the file!
        let mut last_error_time = LOG_THROTTLE.lock().unwrap();
```